chunky/src/java/se/llbit/chunky/resources/ResourcePackLoader.java

```diff
@@ -0,0 +1,200 @@
+        boolean isDefault = pack.equals(MinecraftFinder.getMinecraftJar());
+        return String.format("%s (%s)",
+                isDefault ? "default resource pack" : "resource pack",
+                pack.getAbsolutePath());
+    }
+
+    /**
+     * Load specific resources from the last used set of resource packs.
+     */
+    public static void loadResources(PackLoader[] loaders) {
+        loadResourcePacks(getResourcePacks().toArray(new String[0]), loaders);
+    }
+
+    /**
+     * Load resources from a single resource pack.
+     *
+     * @return True if this is done loading
+     */
+    public static boolean loadResourcePack(File pack, PackLoader[] loaders) {
+        FileSystem resourcePack = null;
+        try {
+            Path root;
+            if (pack.isDirectory()) {
+                // Raw directory
+                resourcePack = FileSystems.getDefault();
+                root = pack.toPath();
+            } else {
+                // Jar or Zip file
+                resourcePack = FileSystems.newFileSystem(URI.create("jar:" + pack.toURI()), Collections.emptyMap());
+                root = resourcePack.getPath("/");
+            }
+
+            boolean complete = true;
+            for (PackLoader loader : loaders) {
+                complete &= loader.load(root, pack.getName());
+            }
+            return complete;
+        } catch (IOException e) {
+            Log.warnf("Failed to open %s: %s", resourcePackName(pack), e.getMessage());
+        } finally {
+            if (resourcePack != null) {
+                try {
+                    resourcePack.close();
+                } catch (UnsupportedOperationException | IOException ignore) {}
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Load all resources in the default set of loaders.
+     */
+    public static void loadResourcePacks(String[] resourcePacks) {
+        // Save the last used set of resource packs
+        TextureCache.reset();
+        ResourcePackLoader.resourcePacks = resourcePacks;
+
+        // Create the loaders
+        PackLoader[] loaders = PACK_LOADER_FACTORIES
+                .stream()
+                .map(PackLoaderFactory::create)
+                .toArray(PackLoader[]::new);
+
+        loadResourcePacks(resourcePacks, loaders);
+    }
+
+    private static void loadResourcePacks(String[] resourcePacks, PackLoader[] loaders) {
+        boolean complete = false;
+        for (String path : resourcePacks) {
+            if (!path.isEmpty()) {
+                File file = new File(path);
+                if (!file.isFile() && !file.isDirectory()) {
+                    Log.errorf("Could not open texture pack: %s", file.getAbsolutePath());
+                } else {
+                    Log.infof("Loading resources from %s", file.getAbsolutePath());
+                    if (loadResourcePack(file, loaders)) {
+                        complete = true;
+                        break;
```

Comment:
Would this mean that if one resource pack contains all textures, it would not load anything that comes after it (for example data packs)?

Reply:
This will only stop loading once all the loaders report they are done loading. The texture pack loader will finish once it loads all textures, the JSON biome loader will never say its done loading.